src/connectors/ProfileServiceConnector.js

```diff
@@ -94,4 +97,186 @@ async function sendActivationQuery(user, token){
+    }
+    return resultData;
+
+}
+export {GetProfileById}
+
+async function UpdateUserProfile(userId, gender, price, phoneNumber, firstName, name, username, email, city, plz, street, houseNumber, token, searchedSkills, achievedSkills){
+    if (price == "" || gender == "" || phoneNumber == "" || firstName == "" || name == "" || username == "" || email == "" || city == "" || plz == "" || street == "" || houseNumber == "") {
+        return
+    }
+    let priceNumber = parseInt(price)
+    if (isNaN(priceNumber)){
+        return "Price is not valid!"
+    }
+    let plzNumber = parseInt(plz)
+    if (isNaN(plzNumber)){
+        return "PLZ is not valid!"
+    }
+    let phoneNr = parseInt(phoneNumber)
+    if (isNaN(phoneNr)){
+        return "Phone Number is not valid!"
+    }
+
+    /*let genderNr
+    switch (gender){
+        case "Weiblich":
+            genderNr = 2
+            break
+        case "Männlich":
+            genderNr = 1
+            break
+        case "Divers":
+            genderNr = 3
+            break
+    }*/
+
+    let query = "http://"+ ipv4 +":8080/profile/" + userId
+    if (searchedSkills == "" & achievedSkills == ""){
+        let user = {
+            "firstName": firstName,
+            "name": name,
+            "gender": gender,
+            "username": username,
+            "email": email,
+            "street": street,
+            "houseNumber": houseNumber,
+            "telephoneNumber": phoneNumber,
+            "price": priceNumber,
+            "city": {
+                "plz": plzNumber,
+                "place": city
+            }
+        }
+    } else {
+        let user = {
+            "firstName": firstName,
+            "name": name,
+            "gender": gender,
+            "username": username,
+            "email": email,
+            "street": street,
+            "houseNumber": houseNumber,
+            "telephoneNumber": phoneNumber,
+            "price": priceNumber,
+            "city": {
+                "plz": plzNumber,
+                "place": city
+            },
+            "achievedSkills": achievedSkills,
+            "searchedSkills": searchedSkills
+        }
+    }
+    const response = await fetch(query, {
+        method: 'PUT',
+        headers: {
+            'Content-type': 'application/json',
+            'Authorization': 'Bearer ' + token
+        },
+        body: JSON.stringify(user)
+    });
+    const resultData = await response.json();
+
+    if (response.status != 200){
+        return resultData.error;
+    }
+    return resultData;
+}
+export {UpdateUserProfile};
+
+async function AddSkill(skillName, level){
+    if (skillName == "" || level == "") {
+        return
+    }
+
+    let query = "http://"+ ipv4 +":8080/skill/"
+    let skill = {
+        "name": skillName,
+        "level": level
+    }
+    const response = await fetch(query, {
+        method: 'PUT',
+        headers: {
+            'Content-type': 'application/json'
```

Comment:
Wird ziemlich sicher einen JWT brauchen. Und wo wird der hinzugefügt? Wenn ein neuer Angelegt wird, dann soll das Frotend das gar nicht können

Reply:
Function not needed anymore anyways

Reply:
Will delete